geopandas/array.py

```diff
@@ -362,62 +435,119 @@ def shape(self):
         -------
         shape : tuple
         """
-
         return tuple([len(self)])
 
-    def ravel(self, order='C'):
-        """ Return a flattened (numpy) array.
+    def to_dense(self):
+        """Return my 'dense' representation
 
         For internal compatibility with numpy arrays.
 
         Returns
         -------
-        raveled : numpy array
+        dense : array
         """
-        return np.array(self)
+        return to_shapely(self.data)
 
-    def view(self):
-        """Return a view of myself.
+    def isna(self):
+        """
+        Boolean NumPy array indicating if each value is missing
+        """
+        return self.data == 0
 
-        For internal compatibility with numpy arrays.
+    def unique(self):
+        """Compute the ExtensionArray of unique values.
 
         Returns
         -------
-        view : Categorical
-           Returns `self`!
+        uniques : ExtensionArray
         """
-        return self
+        from pandas import factorize
+        _, uniques = factorize(self)
+        return uniques
 
-    def to_dense(self):
-        """Return my 'dense' representation
+    @property
+    def nbytes(self):
+        return self.data.nbytes
```

Comment:
This is probably an under-estimate.  Geometries are generally far larger than the pointers.  It's possible that GEOS has something for this, although I doubt it.  If it's cheap we might consider converting to WKB and taking the length of that as an approximation.